relay-general/src/protocol/security_report.rs

```diff
@@ -536,10 +536,14 @@ struct ExpectCtRaw {
     date_time: Option<DateTime<Utc>>,
     hostname: String,
     port: Option<i64>,
+    scheme: Option<String>,
+    #[serde(with = "serde_date_time_3339")]
     effective_expiration_date: Option<DateTime<Utc>>,
     served_certificate_chain: Option<Vec<String>>,
     validated_certificate_chain: Option<Vec<String>>,
     scts: Option<Vec<SingleCertificateTimestampRaw>>,
+    failure_mode: Option<String>,
```

Comment:
The failure mode is listed as mandatory in #843, can we make it non-optional or are some browsers actually omitting it?

Reply:
We went for all fields as optional. I don't think there is a downside to this. When we get a security report we'll collect from it what we can and create an event. Why would we reject a report and not create an event from it when a user agent doesn't send us a particular field?